parts/windows/kuberneteswindowsfunctions.ps1

```diff
@@ -45,7 +45,7 @@ function DownloadFileOverHttp {
         $downloadTimer = [System.Diagnostics.Stopwatch]::StartNew()
         curl.exe -f --retry 5 --retry-delay 0 -L $Url -o $DestinationPath
         if (-not $?) {

```

Comment:
It is safer to use `if ($LASTEXITCODE)` here instead of `if (-not $?)`.
curl can sometimes write to stderr with non-error messages which will cause trigger a false failure.
You might not be hitting this case if all artifacts are currently on the VHD.
We hit an issue here on aks-engine when testing with the non-aks-engine VHDs (for Windows Server Ver 20H2)

Reply:
Thanks for the sharing, @marosset . I will change it